src/components/pages/vendor.tsx

```diff
@@ -1,23 +1,33 @@
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 
 import { Input } from '@/components/atoms/input.tsx'
 import { Typography } from '@/components/atoms/typography.tsx'
-import VendorTable, { Vendor } from '@/components/features/vendor-table.tsx'
+import VendorTable from '@/components/features/vendor-table.tsx'
 import { Footer } from '@/components/molecules/footer.tsx'
 import PageHeader from '@/components/molecules/page-header.tsx'
+import { useGetVendorsQuery } from '@/lib/redux/features/vendor/api.ts'
+import { useCommonStore } from '@/lib/zustand/common.ts'
 
-const VendorPage = () => {
+export default function VendorPage() {
+  const [page, setPage] = useState<number>(1)
   const [productFilter, setProductFilter] = useState('')
   const [locationFilter, setLocationFilter] = useState('')
+  const { setShowLoadingOverlay } = useCommonStore()
+  const { vendors, metadata, isSuccess } = useGetVendorsQuery(
+    { page, product: productFilter, location: locationFilter },
+    {
+      selectFromResult: result => ({
+        ...result,
+        vendors: result.data?.Vendors,
+        metadata: result.data?.Metadata,
+      }),
+    },
+  )
+
+  useEffect(() => {
+    setShowLoadingOverlay(!isSuccess)
+  }, [isSuccess, setShowLoadingOverlay])
```

Comment:
Utility to show global loading overlay can be seen here (with `data-testid` = `loading-overlay`, this can be used when you want to check if the data is no longer loading)